Signup: Replace usage of ThemeThumbnail for ThemesList component on Theme Selection Signup Step

oskosk opened the pull request:
ThemesList is a reusable component appropriate for the function that ThemeSelection Step is providing.

ThemeThumbnail will no longer be necessary. The theme selection step still handles a headStart flow providing {theme: 'pub/twentyfifteen'} as a default dependency.

See #256 and #674 .

cc @scruffian 

## Testing instructions

### Regular theme selection

* Load http://calypso.localhost:3000/start (you'll be redirected to /start/themes)
* Select a theme.
* Follow domain registration (and user signup if testing anonymously). 
* After completing registration, go to switch sites, find your new site.
* Go to themes and check the theme you selected is the one active for your new site.

### Headstart flow

* Load http://calypso.localhost:3000/start/headstart (you'll be redirected to /start/themes)
* Don't select any theme and click the `Skip` button. (This way, theme `twentyfifteen` will be selected by default.
* Follow domain registration (and user signup if testing anonymously). 
* After completing registration, go to switch sites, find your new site.
* Go to themes and check the active theme for your new site is `Twenty Fifteen`.

### /design theme selection for existing sites

* Once logged in, go to switch sites, find one of your sites.
* On sidebar, click Themes under _Personalize_. 
* The new theme gallery is using `ThemesList` and thus, the updated `Theme` component. There shouldn't be any differences with its current behaviour and appearance as the default `Theme` behaviour is used here.

scruffian commented on client/signup/steps/theme-selection/style.scss:
```diff
@@ -1,44 +1,8 @@
-	padding: 12px 0;
-	text-align: center;
+.themes-list .theme__active-focus {
+	opacity: 0.0;
```
Two things. Firstly we shouldn't use opacity to hide stuff, secondly, we don't need the `.0`.

oskosk replied:
Ok, will proceed to update the `Theme` component to show the preview depending on a boolean `.prop`

folletto replied:
Third, any CSS change needs to be namespaced properly for the context it's being used on: this change is being applied everywhere! :)

oskosk replied:
Mhh, sorry for that! This css rule won't stay there as I'll update the `Theme` component to do what this line was supposed to do. Thanks

scruffian commented:
cc @folletto - this changes the theme thumbnails from this:
<img width="1061" alt="screenshot 2015-12-01 11 17 49" src="https://cloud.githubusercontent.com/assets/275961/11499342/333b8b28-981d-11e5-9915-8113f60e4454.png">

to:
<img width="1031" alt="screenshot 2015-12-01 11 17 56" src="https://cloud.githubusercontent.com/assets/275961/11499351/3f7e5000-981d-11e5-9662-1ab1f3156993.png">

Does that look ok to you?

scruffian commented on client/signup/steps/theme-selection/index.jsx:
```diff
@@ -30,28 +34,55 @@ module.exports = React.createClass( {
 	},
 
-	renderThemes: function() {
+	handleScreenshotClick: function( theme ) {
```
I assume this code is from the old theme thumbnail component? Can we remove that at the same time?

oskosk replied:
I don't fully get what you meant. `handlScreenshotClick` is passed as prop to `ThemesList` and forwarded to `Theme` in order to instruct what to do when a screenshot is clicked. In this case this method submits the step.

scruffian commented on client/signup/steps/theme-selection/index.jsx:
```diff
@@ -30,28 +34,55 @@ module.exports = React.createClass( {
+	},
+
+	renderThemesList: function() {
+		var themes = this.props.themes.map( function( t ) {
```
We generally prefer readable names, like `theme` rather than `t`.

oskosk replied:
ok, will fix this

scruffian commented:
Looking good!

folletto commented:
Heya!

Yes, style wise it's correct, however there are two issues:

1. The overlay has disappeared (see the `opacity: 0.0` note above). No CSS should be added at the general namespacing level if it's to be used only in a specific section (in this case, the funnel).
2. The overlay should thus stay there, with just a change of label. As a first proposal for the label, I'd go for: "Pick".

![screen shot 2015-12-01 at 11 52 53](https://cloud.githubusercontent.com/assets/4389/11499994/1717b480-9822-11e5-8d99-6c56d04489d9.png)

oskosk commented:
Awesome @folletto, thanks! @scruffian I'll make the `Theme` component overlay text depend on a prop instead of hiding/showing it depending on a boolean prop. Defaulting to `'preview'`.

oskosk commented:
* Updated Theme component to let you choose the actionLabel text
* Updated theme-selection to use 'Pick' text for the label.

cc @scruffian @folletto 
 
Before this pull request:
<img width="1061" alt="screenshot 2015-12-01 11 17 49" src="https://cloud.githubusercontent.com/assets/275961/11499342/333b8b28-981d-11e5-9915-8113f60e4454.png">

After
<img width="1033" alt="screen shot 2015-12-01 at 10 08 57 am" src="https://cloud.githubusercontent.com/assets/746152/11501554/d955870c-9813-11e5-8997-961b6d683d48.png">

folletto commented:
Design wise :+1:

scruffian commented:
LGTM :+1:

scruffian commented:
As a followup it would be good to use the same component in the DSS flow...
